mailet/standard/src/main/java/org/apache/james/transport/mailets/StripAttachment.java

```diff
@@ -324,21 +332,22 @@ private boolean isMatching(BodyPart bodyPart, String fileName) throws MessagingE
 
     private void storeBodyPartAsFile(BodyPart bodyPart, Mail mail, String fileName) throws Exception {
         if (directoryName != null) {
-            Optional<String> filename = saveAttachmentToFile(bodyPart, Optional.of(fileName));
-            if (filename.isPresent()) {
-                addFilenameToAttribute(mail, filename.get(), SAVED_ATTACHMENTS_ATTRIBUTE_KEY);
-            }
+            saveAttachmentToFile(bodyPart, Optional.of(fileName)).ifPresent(filename ->
+                    addFilenameToAttribute(mail, AttributeValue.of(filename), SAVED_ATTACHMENTS)
+            );
         }
     }
 
-    private void addFilenameToAttribute(Mail mail, String filename, String attributeName) {
-        @SuppressWarnings("unchecked")
-        List<String> attributeValues = (List<String>) mail.getAttribute(attributeName);
-        if (attributeValues == null) {
-            attributeValues = new ArrayList<>();
-            mail.setAttribute(attributeName, (Serializable) attributeValues);
-        }
-        attributeValues.add(filename);
+    private void addFilenameToAttribute(Mail mail, AttributeValue<String> attributeValue, AttributeName attributeName) {
+        Function<List<AttributeValue<String>>, List<AttributeValue<?>>> typeWeakner = values ->
+                values.stream().map(value -> (AttributeValue<?>) value).collect(Collectors.toList());
+
+        List<AttributeValue<?>> attributeValues = AttributeUtils
+                .getValueAndCastFromMail(mail, attributeName, LIST_OF_STRINGS)
+                .map(typeWeakner)
+                .orElse(new ArrayList<>());
```

Comment:
`ImmutableList.of()` ?

Reply:
No, we do a `attributeValues.add(attributeValue);` the line below